# Microsoft Studios Movie Project


# 1. Introduction

## 1.1 Overview

Films are a major part of popular culture and a huge source of entertainment for many people. The market is projected to reach USD 409.02 billion by 2026 in terms of revenue. Therefore, it is wise to invest in this market. I will use data from various websites that contain information about the film industry.

## 1.2 Objectives

1. Business Understanding<br>
2. Data Understanding <br>
3. Data Preparation
4. Data Analysis
5. Conclusion

# 2. Business Understanding

The tech giant Microsoft has decided to venture into creating original video content and wants to establish a movie studio. My goal is to use exploratory data analysis to produce insights for Microsoft as they enter the film industry. I will be looking for answers to the following questions:

1.What are the most profitable genres, franchises, and stars in each market and how do they differ across regions?

2.How do critical and audience ratings and reviews affect the box office performance of different films and how can they be used to predict demand and optimize marketing strategies?

3.What are the current and future challenges and opportunities in the film industry and how can Microsoft leverage its strengths and resources to create original video content that meets the need of the users?


# 3. Data Understanding

The datasets provided for this analysis were collected from different movie review aggregation sites and contain information on the various movie genres and their popularity among critics and viewers.  
The datasets include:
1. [IMDB](https://www.imdb.com/) 
2. [Box Office Mojo](https://www.boxofficemojo.com/) 
3. [Rotten Tomatoes](https://www.rottentomatoes.com/)


## Steps
1. Load the data with pandas and explore the dataframes.
2. Clean the data by dealing with:
    - missing values
    - duplicate rows
    - invalid data
    - outliers
3. Perform exploratory analysis in order to answer the business questions.
4. Conclusion.
5. Recommendations.

# 3.1 Loading Libraries and Datasets

In [3]:
# importing the packages I will be using for this project
import numpy as np
import sqlite3
import pandas as pd
import zipfile
import csv
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [8]:
#loading the datasets
#first extract the im.db file and create a database connection
zf = zipfile.ZipFile('zippedData/im.db.zip')
zf.extract('im.db')
conn = sqlite3.connect('im.db')
#Loading other datasets
rt_reviews = pd.read_csv('zippedData/rt.reviews.tsv.gz',delimiter = "\t",encoding='latin-1')
rt_movies = pd.read_csv('zippedData/rt.movie_info.tsv.gz',delimiter = '\t')
bom_movies = pd.read_csv('zippedData/bom.movie_gross.csv.gz')


# 3.2 Previewing the Datasets

#### a. bom_movies

In [11]:
#previewing the top
bom_movies.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [13]:
#previewing the bottom 
bom_movies.tail()

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [14]:
#determining the number of rows and columns
bom_movies.shape

(3387, 5)

In [15]:
#previewing bom_movies information
bom_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [16]:
#previewing the summary statistics of bom_movies
bom_movies.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

Observations:

1.The mean domestic gross is about 28.7 million dollars, with a large standard deviation of about 67 million dollars. The minimum domestic gross is 100 dollars and the maximum is about 936.7 million dollars.

2.The DataFrame has 3387 rows and 5 columns.

3.The info method shows that there are some missing values in the studio, domestic_gross, and foreign_gross columns

4.The foreign_gross column is of the object data type, which suggests that it may contain non-numeric values.